src/fixedSizeSquareMatrixOpsImpl.hpp

```diff
@@ -88,7 +88,18 @@ struct SquareMatrixOps< 2 >
   static auto determinant( MATRIX const & matrix )
   {
     checkSizes< 2, 2 >( matrix );
-    return matrix[ 0 ][ 0 ] * matrix[ 1 ][ 1 ] - matrix[ 0 ][ 1 ] * matrix[ 1 ][ 0 ];
+
+    // Aliases for matrix [[a, b],[c, d]] for improved readability
+    auto const & a = matrix[0][0];
+    auto const & b = matrix[0][1];
+    auto const & c = matrix[1][0];
+    auto const & d = matrix[1][1];
+
+    // Using the more precise Kahan method to compute the 2x2 determinant.
+    auto const w = b * c;
+    auto const e = fma( -b, c, w );
```

Comment:
Is the `fma` required to obtain adequate precision? I'd prefer to do `auto const e = -b * c + w` and let the compiler do it's thing.

Reply:
Sure, I'll check too.

Reply:
https://godbolt.org/z/ao9GMhPsq